cmd/clusterctl/client/cluster/upgrader_info.go

```diff
@@ -72,6 +72,12 @@ func (u *providerUpgrader) getUpgradeInfo(provider clusterctlv1.Provider) (*upgr
 		if err != nil {
 			return nil, errors.Wrapf(err, "failed to parse available version for the %s provider", provider.InstanceName())
 		}
+
+		// ignore pre-releases when getting upgrade info
+		if availableSemVersion.PreRelease() != "" {
```

Comment:
I'm a little bit debated about excluding pre-releases from upgrades for two reasons:
- While in clusterctl init the users get pre-release installed without any prior info, for upgrades there is the `upgrade plan` command that informs the user about the target version, so the user can choose willingly if to install a pre-release version or not
- If we are excluding them, in case the user wants to install a pre-release the UX in not intuitive and it should be documented (providers should be listed one by one like in custom upgrades)

Focusing on the specific change, might be we should continue to read all the version in this method and filter out pre-releases in `getLatestNextVersion` only, so info about pre-release will be available if required

Reply:
Ohh, I just saw the comment on the issue.
So ok with the idea to not propose pre-release in upgrade plans, but please explore if this can be achieved by changing `getLatestNextVersion` instead of  `getUpgradeInfo`

Reply:
Sure. I'll add some documentation on upgrading to pre-releases.
I'll take a look at the `getLatestNextVersion` vs `getUpgradeInfo`. Thanks!